dsgrid/registry/filter_registry_manager.py

```diff
@@ -0,0 +1,96 @@
+import logging
+
+from dsgrid.config.simple_models import RegistrySimpleModel
+from dsgrid.config.dataset_schema_handler_factory import make_dataset_schema_handler
+from dsgrid.utils.timing import track_timing, timer_stats_collector
+from .registry_manager import RegistryManager
+
+
+logger = logging.getLogger(__name__)
+
+
+class FilterRegistryManager(RegistryManager):
+    """Specialized RegistryManager that performs filtering operations."""
+
+    @track_timing(timer_stats_collector)
+    def filter(self, simple_model: RegistrySimpleModel):
+        """Filter the registry as described by simple_model.
+
+        Parameters
+        ----------
+        simple_model : RegistrySimpleModel
+            Filter all configs and data according to this model.
+
+        """
+        project_ids_to_keep = {x.project_id for x in simple_model.projects}
+        to_remove = [x for x in self._project_mgr.list_ids() if x not in project_ids_to_keep]
+        for project_id in to_remove:
+            self._project_mgr.remove(project_id)
+
+        dataset_ids_to_keep = {x.dataset_id for x in simple_model.datasets}
+        to_remove = [x for x in self._dataset_mgr.list_ids() if x not in dataset_ids_to_keep]
+        for dataset_id in to_remove:
+            self._dataset_mgr.remove(dataset_id)
+
+        modified_dims = set()
+        modified_dim_records = {}
+
+        # Note: Use pandas to write CSVs because Spark produces directories.
+
+        def handle_dimension(simple_dim, dim):
+            records = dim.get_records_dataframe()
+            filename = dim.src_dir / dim.model.filename
+            df = records.filter(records.id.isin(simple_dim.record_ids))
+            df.toPandas().to_csv(filename, index=False)
+            modified_dims.add(dim.model.dimension_id)
+            modified_dim_records[dim.model.dimension_id] = {
+                x.id for x in df.select("id").distinct().collect()
```

Comment:
Didn't you come up with a faster way to turn distinct value collection into a list? I can't remember.

Reply:
Yes, and it's a bit awkward. I couldn't remember and it doesn't matter in this case because the records are small.